# Final Project: Part 1

Read and explore the dataset.

In [2]:
import pandas as pd

In [6]:
# read the dataset
df = pd.read_csv('Daily_Clery_Crime_Log.csv')

In [7]:
df.head()

Incident Date reported Time reported Date occurred Time occurred  \
0  II1700001      1/1/2017          2:15      1/1/2017          2:14   
1  II1700001      1/1/2017          2:15      1/1/2017          2:14   
2  II1700001      1/1/2017          2:15      1/1/2017          2:14   
3  IF1700001      1/1/2017          0:43      1/1/2017          0:43   
4  IF1700002      1/1/2017          1:19      1/1/2017          1:19   

         General Location                 Crime Description   Disposition  
0      GREEN E / FIRST ST    DRIVING UNDER THE INFL-ALCOHOL        ARREST  
1      GREEN E / FIRST ST  CANNABIS POSS 10 GM OR LESS -ORD       PENDING  
2      GREEN E / FIRST ST         BREATH ALCOHOL OVER LIMIT        ARREST  
3            51 E GREGORY                             FIGHT  ADMIN CLOSED  
4  SHERMAN RESIDENCE HALL                 LIQUOR - UNDERAGE  ADMIN CLOSED

In [9]:
df.tail()

Incident Date reported Time reported Date occurred Time occurred  \
4602  00790-2018     3/13/2018         14:23     3/11/2018         23:35   
4603  00795-2018     3/13/2018         14:23      3/8/2018         20:11   
4604   CSA18-064     3/13/2018          9:37     3/10/2018       UNKNOWN   
4605   CH1802052     3/13/2018          6:00     3/13/2018          0:40   
4606   IT1800171     3/14/2018          2:02     3/14/2018          2:02   

            General Location                 Crime Description  \
4602  WASSAJA RESIDENCE HALL                             DRUGS   
4603                    SDRP                           LARCENY   
4604              OFF CAMPUS                              RAPE   
4605          306 E GREEN ST                DISORDERLY CONDUCT   
4606  GREEN ST / GOODWIN AVE  CANNABIS POSS 10 GM OR LESS -ORD   

                   Disposition  
4602              ADMIN CLOSED  
4603              ADMIN CLOSED  
4604              ADMIN CLOSED  
4605  REPORTED TO OTHER AGENCY  
4606                    ARREST

In [10]:
len(df)

4607

In [11]:
df.describe()

Incident Date reported Time reported Date occurred Time occurred  \
count        4607          4607          4607          4607          4607   
unique       3841           402          1051           640           927   
top     UU1705578     3/14/2017          0:00      3/3/2017       UNKNOWN   
freq           11           124          1929            49           213   

       General Location     Crime Description               Disposition  
count              4607                  4607                      4607  
unique             1296                   288                        15  
top            RED LION  UNLAWFUL USE OF I.D.  REPORTED TO OTHER AGENCY  
freq                281                   412                      1972

In [30]:
crime_counts = df['Date reported'].value_counts()

In [40]:
crime_counts

3/14/2017     124
2/22/2017     111
3/30/2017      75
10/31/2017     74
4/27/2017      67
2/1/2017       66
3/29/2017      63
11/8/2017      62
7/13/2017      59
5/18/2017      55
9/26/2017      53
6/16/2017      51
1/24/2017      49
5/9/2017       46
2/1/2018       46
10/12/2017     42
7/23/2017      41
5/4/2017       40
10/13/2017     37
11/10/2017     36
11/16/2017     36
2/16/2018      35
6/21/2017      32
10/2/2017      31
9/15/2017      30
8/23/2017      29
3/23/2017      29
4/14/2017      29
9/13/2017      28
3/6/2018       28
             ... 
2/25/2017       2
8/7/2017        2
1/14/2017       1
12/26/2017      1
6/14/2017       1
3/13/2017       1
7/10/2017       1
8/2/2017        1
3/18/2017       1
3/14/2018       1
11/19/2017      1
3/19/2017       1
12/24/2017      1
12/28/2017      1
11/24/2017      1
11/22/2017      1
6/24/2017       1
7/16/2017       1
1/5/2017        1
1/12/2017       1
8/15/2017       1
5/28/2017       1
12/31/2017      1
2/11/2018       1
5/21/2017 

In [34]:
df.dtypes

Incident             object
Date reported        object
Time reported        object
Date occurred        object
Time occurred        object
General Location     object
Crime Description    object
Disposition          object
dtype: object

In [35]:
df.index

RangeIndex(start=0, stop=4607, step=1)

In [37]:
data.shape

(4607, 8)

In [59]:
df['Crime Description'].head(10)

0      DRIVING UNDER THE INFL-ALCOHOL
1    CANNABIS POSS 10 GM OR LESS -ORD
2           BREATH ALCOHOL OVER LIMIT
3                               FIGHT
4                   LIQUOR - UNDERAGE
5                     THEFT OVER $500
6                    WARRANT-IN STATE
7                THEFT $500 AND UNDER
8                      SEX HARASSMENT
9                     THEFT OVER $500
Name: Crime Description, dtype: object

In [43]:
time_series = pd.DataFrame(df['Date reported'].value_counts().reset_index())
time_series.columns = ['date', 'count']

In [56]:
new_ts = time_series.sort_values(by = ['date'])

In [58]:
new_ts.plot(x='date', y='count')
# plt.show()

The dataset I used is named "Daily_Clery_Crime_Log". It represent the daily clery crime log in our campus from the entire 2017 year and January to March 14th in 2018. We obtained it from a record clerk in Police department, and here is a URL to obatain it: http://police.illinois.edu/crime-reporting/daily-crime-log/. But the data in this website is real-time updated and only reserve latest 60 days records. The data is no license so we can do anything we want to do such as explore the data, clean and modify the data, visualize the data, etc. After I explored the data initially, I found that the data has 4607 rows and eight columns. The rows are represent crime event records sorted by the date, and the columns represents the ID of iccident, reported date, reported time, occurred date, occurred time, general location, crime description and the disposition. The specific crime description includes: driving under the infl-alcohol, fight, sex harassment, theft and so on. When I used df.dtypes() to find the data types, I found all the datatypes in each column is object. And I also found that most crime event occurred in Mar 14th 2017 by some simple manipulations of dataset.